api/catalog/api/models/image.py

```diff
@@ -48,41 +49,43 @@ class Image(ImageFileMixin, AbstractMedia):
     class Meta(AbstractMedia.Meta):
         db_table = "image"
 
+    @property
+    def mature(self) -> bool:
+        return MatureImage.objects.filter(identifier=self.identifier).exists()
```

Comment:
Based on my understanding, the `@cached_property` decorator wouldn't not do much here, since that it's lifetime is bound to the lifetime of the model instance?

Django has a decorator with the same name, and their [docs](https://docs.djangoproject.com/en/4.1/ref/utils/#django.utils.functional.cached_property) state that the cache only persists as long as the instance is in memory.